File: backend/src/controllers/integrations/ClaudeController.js
import {container} from '../../services/container'

const claudeService = container.get('claudeService')

const ClaudeController = {
  sendMessages: async ctx => {
    try {
      const {messages, model, ...params} = await ctx.request.json('infinity')
      const apiKey = ctx.headers['x-api-key']

      if (!apiKey) {
        ctx.throw(400, 'Claude API key not found')
      }

      if (!model) {
        ctx.throw(400, 'Model name not specified')
      }

      if (!messages || messages.length === 0 || !messages[0]?.content) {
        ctx.throw(400, 'Messages not specified')
      }

      if (!params.max_tokens) {
        ctx.throw(400, 'max_tokens not specified')
      }

      const result = await claudeService.sendMessages({
        model,
        messages,
        ...params,
      })

      ctx.body = result
    } catch (e) {
      ctx.throw(e.status || 500, e.message || 'Internal Server Error')
    }
  },
}

export default ClaudeController

File: backend-v2/internal/modules/integration/claude_controller.go
package integration

import (
	"backend-v2/internal/common/response"
	"backend-v2/internal/services/claude"
	"fmt"

	"github.com/gofiber/fiber/v2"
	"github.com/qiniu/qmgo"
)

/* ClaudeController handles Claude AI-specific endpoints */
type ClaudeController struct {
	service claude.Service
	db      *qmgo.Database
}

func NewClaudeController(service claude.Service, db *qmgo.Database) *ClaudeController {
	return &ClaudeController{service: service, db: db}
}

/* ClaudeMessagesRequest holds Claude messages API request data */
type ClaudeMessagesRequest struct {
	Messages  []claude.Message `json:"messages"`
	Model     string           `json:"model"`
	MaxTokens int              `json:"max_tokens"`
}

func (r *ClaudeMessagesRequest) Validate() error {
	if len(r.Messages) == 0 {
		return fmt.Errorf("Messages not specified")
	}
	if r.Model == "" {
		return fmt.Errorf("Model name not specified")
	}
	if r.MaxTokens == 0 {
		return fmt.Errorf("max_tokens not specified")
	}
	return nil
}

/* Messages handles Claude messages API */
func (ctrl *ClaudeController) Messages(c *fiber.Ctx) error {
	userID, err := getUserID(c)
	if err != nil {
		return err
	}

	var req ClaudeMessagesRequest
	if err := parseBody(c, &req); err != nil {
		return err
	}

	if err := req.Validate(); err != nil {
		return response.BadRequest(c, err.Error())
	}

	result, err := ctrl.service.Messages(ctrl.db, userID, req.Messages, req.Model, req.MaxTokens)
	if err != nil {
		return response.InternalError(c, err.Error())
	}

	return c.JSON(result)
}
